kingpin/actors/aws/ecs.py

```diff
@@ -366,87 +422,528 @@ def _tasks_done(self, tasks):
+        this will raise an exception.
+
+        Args:
+            service_name: name of the service to wait for.
+            task_definition_name: Task Definition string.
+        """
+        # Create set used to ensure event logs are only printed once.
+        self.seen_events = set()
+        while True:
+            done = yield self._is_service_deployed(
+                service_name, task_definition_name)
+            if done:
+                break
+            yield gen.sleep(10)
+
+    @gen.coroutine
+    def _is_service_deployed(self, service_name, task_definition_name):
+        """Checks if service is finished deploying.
+        Meant to be called in a wait-loop.
+
+        Args:
+            service_name: name of the service to wait for.
+            task_definition_name: Task Definition string.
+
+        Returns:
+            A boolean indicating whether the service is completely deployed.
+        """
+        service = yield self._describe_service(service_name)
+
+        deployments = service['deployments']
+        primary_deployment = None
+        for deployment in deployments:
+            if deployment['status'] == 'PRIMARY':
+                primary_deployment = deployment
+                break
+        if not primary_deployment:
+            # There should always be one 'PRIMARY' deployment returned.
+            raise exceptions.RecoverableActorFailure(
+                'No primary deployment.')
+
+        # Check that the primary deployment is for the correct task definition.
+        primary_task_definition_name = self._arn_to_name(
+            primary_deployment['taskDefinition'])
+        if primary_task_definition_name != task_definition_name:
+            raise exceptions.RecoverableActorFailure(
+                'Primary deployment was for {}, not {}.'.format(
+                    primary_task_definition_name,
+                    task_definition_name))
+
+        service_timestamp = primary_deployment['createdAt']
+
+        sorted_new_events = self._get_sorted_new_log_events(
+            events=service['events'],
+            start_timestamp=service_timestamp)
+
+        for event in sorted_new_events:
+            event_timestamp, event_message = event
+            self.log.info('Event: {message}'.format(
+                message=event_message))
+
+        running_count = primary_deployment['runningCount']
+        desired_count = primary_deployment['desiredCount']
+        missing_count = desired_count - running_count
+
+        extra_deployment_count = len(deployments) - 1
+
+        if missing_count == 0 and extra_deployment_count == 0:
+            self.log.info('Service {name} deployed successfully.'.format(
+                name=service_name))
+            raise gen.Return(True)
+
+        self.log.info(
+            '{running} tasks running out of {desired}, '
+            'and {extra} deployments waiting on termination'.format(
+                running=running_count,
```

Comment:
This is a tiny nit -- but you bounce around in syntax here. In other places you use `{}`, and here you use `{running}`. I have no preference, but I'd like it to be consistent at least throughout the same module.